carmin-server/server/resources/helpers/executions.py

```diff
@@ -139,3 +134,27 @@ def validate_request_model(model: dict,
             INVALID_INPUT_FILE, error)
         return False, error_code_and_message
     return True, None
+
+
+def filter_executions(executions, offset, limit):
+    if offset:
+        try:
```

Comment:
Please extract this validation in the calling function. This should only accept int offset and limit.

Reply:
This was attempted but is very impractical. I found that the `args.get(arg, default, type)` function does not easily allow this kind of checking. If the `arg` cannot be converted to `type`, `arg` is then given the default value without an error being raised. This is problematic since there is no way to differentiate between a conversion error (text `string` to `int`, which is an invalid query), and a request without an `offset` or `limit` (which is valid).

In order to keep the error handling provided by `filter_executions`, it will not be changed to offload the responsibility of null checking to the caller.